src/functions.php

```diff
@@ -77,7 +77,7 @@ function uri_for($uri)
 function stream_for($resource = '', array $options = [])
 {
     switch (gettype($resource)) {
-        case 'string':
+        case is_scalar($resource):
```

Comment:
From what I understand I should leave it like it is. Do you want me to add anything else ?